server/src/main/java/org/elasticsearch/common/settings/Setting.java

```diff
@@ -1372,6 +1372,68 @@ public static ByteSizeValue parseByteSize(String s, ByteSizeValue minValue, Byte
         return value;
     }
 
+    public static <T extends Enum<T>> Setting<T> enumSetting(Class<T> clazz, String key, T defaultValue, Property... properties) {
```

Comment:
I wasn't sure wether or not to add all the signatures, which I copied from `boolSetting`. Also wasn't sure about adding comments. The added functionality seems straightforward and use is similar to f.e. `boolSetting`

Reply:
Some basic javadocs would be appreciated. While we are severely lacking them currently, we try to add them with new stuff when possible. Regarding multiple variants: less is better. So I would just add the minimum variants necessary for the existing enum use cases. We can always add more later, but we have a large problem of seldom used methods/ctors, when a single method/ctor would suffice.